ebpf/event_common.go

```diff
@@ -95,22 +94,31 @@ func (c ConnectionStats) ByteKey(buffer *bytes.Buffer) ([]byte, error) {
 	return buffer.Bytes(), nil
 }
 
+// StrKey returns a unique key for this connection represented as a string
+func (c ConnectionStats) StrKey() (string, error) {
+	buf := &bytes.Buffer{}
```

Comment:
There was a reason this was pulled out and reused, it can be pretty expensive to reallocate a buffer every time.

Reply:
https://golang.org/src/bytes/buffer.go

Take a look at: `bootstrap [64]byte // memory to hold first slice; helps small buffers avoid allocation.`. Quite a lot of allocations if you create a new one each iteration.